sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -608,21 +637,82 @@ trait DataFrame extends RDDApi[Row] {
   def saveAsTable(
       tableName: String,
       dataSourceName: String,
+      mode: SaveMode,
       options: java.util.Map[String, String]): Unit
 
+  /**
+   * :: Experimental ::
+   * (Scala-specific)
+   * Creates a table from the the contents of this DataFrame based on a given data source,
+   * [[SaveMode]] specified by mode, and a set of options.
+   *
+   * Note that this currently only works with DataFrames that are created from a HiveContext as
+   * there is no notion of a persisted catalog in a standard SQL context.  Instead you can write
+   * an RDD out to a parquet file, and then register that file as a table.  This "table" can then
+   * be the target of an `insertInto`.
+   */
+  @Experimental
+  def saveAsTable(
```

Comment:
i think u can implement all other versions of saveAsTable here in this interface; then the only one u need to implement is saveAsTable in DataFrameImpl and IncomputableColumn